### PR TITLE
Master relational model fge

### PR DESCRIPTION
**Pull request by FrancoisGe**

### CODE REVIEW

File: addons/web/static/src/views/fields/statusbar/statusbar_field.js [hunk 1]
```diff
@@ -243,6 +210,28 @@ export class StatusBarField extends Component {
...
+            return;
+        }
+
+        const fieldNames = ["display_name"];
+        if (foldField) {
+            fieldNames.push(foldField);
+        }
+
+        let domain = props.domain;
+        if (domain.length && record.data[fieldName]) {
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/statusbar/statusbar_field.js]: remove and add if after in special data if not in it

File: addons/web/static/src/views/relational_model/relational_model.js [hunk 2]
```diff
@@ -93,6 +93,8 @@ export class RelationalModel extends Model {
         this.keepLast = markRaw(new KeepLast());
         this.mutex = markRaw(new Mutex());
 
+        this.loadSpecialData = {};
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/relational_model.js]: export

File: addons/web/static/src/views/fields/radio/radio_field.js [hunk 3]
```diff
@@ -1,44 +1,56 @@
...
-    static getItems(fieldName, record) {
-        switch (record.fields[fieldName].type) {
+    setup() {
+        this.id = `radio_field_${++RadioField.nextId}`;
+
+        const _loadSpecialData = useSpecialData(
+            "selection_badge",
+            async (orm, relation, domain) => {
+                const records = await orm.searchRead(relation, domain, ["id"]);
+                return orm.call(relation, "name_get", [records.map((record) => record.id)]);
```
Comment (FrancoisGe) [inline addons/web/static/src/views/fields/radio/radio_field.js]: searchRead Unity 
display name

File: addons/web/static/src/views/kanban/kanban_controller.js [hunk 4]
```diff
@@ -96,6 +96,7 @@ export class KanbanController extends Component {
             defaultGroupBy,
             defaultOrderBy: archInfo.defaultOrder,
             maxGroupByDepth: 1,
+            loadSpecialData: state?.modelState.loadSpecialData,
```
Comment (FrancoisGe) [inline addons/web/static/src/views/kanban/kanban_controller.js]: import only state

File: addons/web/static/src/views/relational_model/relational_model.js [hunk 5]
```diff
@@ -142,6 +144,13 @@ export class RelationalModel extends Model {
         return this.config;
     }
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model/relational_model.js:150]: remove

File: addons/web/static/src/views/fields/badge_selection/badge_selection_field.js [hunk 6]
```diff
@@ -1,24 +1,34 @@
 /** @odoo-module **/
 
-import { registry } from "@web/core/registry";
+import { Component } from "@odoo/owl";
```
Comment (FrancoisGe) [inline addons/web/static/src/views/fields/badge_selection/badge_selection_field.js:3]: specialData

File: addons/web/static/src/views/fields/radio/radio_field.js [hunk 7]
```diff
@@ -1,44 +1,51 @@
...
         }
     }
-
-    setup() {
-        this.id = `radio_field_${++RadioField.nextId}`;
-    }
-
-    get items() {
-        return RadioField.getItems(this.props.name, this.props.record);
-    }
```
Comment (FrancoisGe) [inline addons/web/static/src/views/fields/radio/radio_field.js:41]: check use case and fix

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**